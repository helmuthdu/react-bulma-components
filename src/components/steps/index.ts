export * from './steps';
export * from './steps-item';
export * from './steps-mark';
