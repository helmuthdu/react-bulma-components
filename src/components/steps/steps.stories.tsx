import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { Steps } from '.';

storiesOf('Steps', module)
  .add('Default', () => <Steps />);
