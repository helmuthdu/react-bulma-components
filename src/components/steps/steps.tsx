import clsx from 'clsx';
import * as React from 'react';
import { Sizes } from '../../constants';
import { Element } from '../../elements';
import { StepsItem } from './steps-item';
import { StepsMark } from './steps-mark';

type StepsProps = React.PropsWithChildren<{
  as?: keyof JSX.IntrinsicElements | React.ComponentType<any>;
  size?: Sizes;
  className?: string;
}>;

export const Steps: React.FunctionComponent<StepsProps> & {
  Item: typeof StepsItem,
  Mark: typeof StepsMark,
} = ({ className, children, size, ...props }: StepsProps) => (
  <Element
    {...props}
    className={clsx('steps', className, {
      [`is-${size}`]: size
    })}>
    {children}
  </Element>
);

Steps.Item = StepsItem;
Steps.Mark = StepsMark;

Steps.defaultProps = {
  as: 'ul'
};

