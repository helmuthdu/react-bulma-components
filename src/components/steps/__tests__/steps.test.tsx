import { render } from '@testing-library/react';
import * as React from 'react';
import { Steps } from '../steps';

describe('Steps component', () => {
  it('should render', () => {
    const { asFragment } = render(<Steps>content</Steps>);
    expect(asFragment()).toMatchSnapshot();
  });
});
