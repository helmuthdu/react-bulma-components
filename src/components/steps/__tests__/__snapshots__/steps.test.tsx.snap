// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps component should render 1`] = `
<DocumentFragment>
  <ul
    class="steps"
  >
    content
  </ul>
</DocumentFragment>
`;
